components/custom-network/token-configuration.tsx

```diff
@@ -55,29 +55,42 @@ export default function TokenConfiguration({
   }
 
   function handleNFTTokenChange(e) {
-    changedDataHandler("tokens", { label: "validated", value: false });
     changedDataHandler("tokens", { label: "nftToken", value: e.target.value });
   }
 
+  function handleNetworkTokenChange(token: Token) {
+    changedDataHandler("tokens", { label: "networkToken", value: token.address });
+  }
+
   function setDeployedAddress(address) {
-    changedDataHandler("tokens", { label: "validated", value: true });
     changedDataHandler("tokens", { label: "nftToken", value: address });
   }
 
-  async function validateAddress() {
+  async function validateNFTAddress() {
     if (data.nftToken.trim() === "") return changedDataHandler("tokens", { label: "validated", value: false });
 
-    try {
-      const token = new BountyToken(BeproService.bepro, data.nftToken);
+    const token = new BountyToken(BeproService.bepro, data.nftToken);
```

Comment:
So, if data.nftToken _is not_ a BountyToken contract, this will throw an error that you're not catching. We need to catch that error and tell it to the user so the user understands _what kind of contract it needs_ because they could copy-paste a NFT address and think it will work